subsys/net/ip/net_shell.c

```diff
@@ -2624,6 +2619,152 @@ int net_shell_cmd_tcp(int argc, char *argv[])
 	return 0;
 }
 
+#if defined(CONFIG_NET_VLAN)
+static void iface_vlan_del_cb(struct net_if *iface, void *user_data)
+{
+	u16_t vlan_tag = POINTER_TO_UINT(user_data);
+	int ret;
+
+	ret = net_eth_vlan_disable(iface, vlan_tag);
+	if (ret < 0) {
+		if (ret != -ESRCH) {
+			printk("Cannot delete VLAN tag %d from interface %p\n",
+			       vlan_tag, iface);
+		}
+
+		return;
+	}
+
+	printk("VLAN tag %d removed from interface %p\n",
+	       vlan_tag, iface);
+}
+
+static void iface_vlan_cb(struct net_if *iface, void *user_data)
+{
+	struct ethernet_context *ctx = net_if_l2_data(iface);
+	int *count = user_data;
+	int i;
+
+	if (net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)) {
+		return;
+	}
+
+	if (*count == 0) {
+		printk("    Interface  Type     Tag\n");
+	}
+
+	if (!ctx->vlan_enabled) {
+		printk("VLAN tag(s) not set\n");
+		return;
+	}
+
+	for (i = 0; i < NET_VLAN_MAX_COUNT; i++) {
+		if (!ctx->vlan[i].iface || ctx->vlan[i].iface != iface) {
+			continue;
+		}
+
+		if (ctx->vlan[i].tag == NET_VLAN_TAG_UNSPEC) {
+			continue;
+		}
+
+		printk("[%d] %p %s %d\n", net_if_get_by_iface(iface), iface,
+		       iface2str(iface, NULL), ctx->vlan[i].tag);
+
+		break;
+	}
+
+	(*count)++;
+}
+#endif /* CONFIG_NET_VLAN */
+
+int net_shell_cmd_vlan(int argc, char *argv[])
+{
+#if defined(CONFIG_NET_VLAN)
+	int arg = 1;
+	int ret;
+	u16_t tag;
+
+	if (argv[arg]) {
```

Comment:
too big if. 

Instead: if (!argv[arg]) { ... do the actual else below at line 2757.. return 0; }
Rest would be on the same level as function block

Reply:
Sure, makes sense.